pilot/helpers/agents/Developer.py

```diff
@@ -143,46 +142,38 @@ def implement_task(self, i, development_task=None):
                 logger.warning('Testing at end of task failed')
                 break
 
-    def replace_old_code_comments(self, files_with_changes):
-        files_with_comments = [{**file, 'comments': [line for line in file['content'].split('\n') if '[OLD CODE]' in line]} for file in files_with_changes]
-
-        for file in files_with_comments:
-            if len(file['comments']) > 0:
-                fully_coded_file_convo = AgentConvo(self)
-                fully_coded_file_response = fully_coded_file_convo.send_message(
-                    'development/get_fully_coded_file.prompt', {
-                        'file': self.project.get_files([file['path']])[0],
-                        'new_file': file,
-                    }, GET_FULLY_CODED_FILE)
-
-                file['content'] = fully_coded_file_response['file_content']
-
-        return files_with_comments
-
     def step_code_change(self, convo, task_description, step, i, test_after_code_changes):
         if 'code_change_description' in step:
-            # TODO this should be refactored so it always uses the same function call
             print(f'Implementing code changes for `{step["code_change_description"]}`')
             code_monkey = CodeMonkey(self.project, self)
-            updated_convo = code_monkey.implement_code_changes(convo, task_description, step['code_change_description'],
-                                                               step, i)
+            updated_convo = code_monkey.implement_code_changes(convo, task_description, step['code_change_description'], step, i)
```

Comment:
This flow happens as a result of debugging (and has no file name/path in the step data).